app/backend/src/couchers/models.py

```diff
@@ -317,19 +317,63 @@ def __repr__(self):
 
 class UserSession(Base):
     """
-    Active session on the app, for auth
+    API keys/session cookies for the app
+
+    There are two types of sessions: long-lived, and short-lived. Long-lived are
+    like when you choose "remember this browser": they will be valid for a long
+    time without the user interacting with the site. Short-lived sessions on the
+    other hand get invalidated quickly if the user does not interact with the
+    site.
+
+    Long-lived tokens are valid from `created` until `expiry`.
+
+    Short-lived tokens expire after 2 hours if they are not used.
     """
 
     __tablename__ = "sessions"
     token = Column(String, primary_key=True)
 
     user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
 
-    # timezone should always be UTC
-    started = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+    # whether it's a long-lived or short-lived session
+    long_lived = Column(Boolean, nullable=False)
+
+    # the time at which the session was created
+    created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+    # the expiry of the session: the session *cannot* be refreshed past this
+    expiry = Column(DateTime(timezone=True), nullable=False, server_default=func.now() + text("interval '7 days'"))
+
+    # the time at which the token was invalidated, allows users to delete sessions
+    deleted = Column(DateTime(timezone=True), nullable=True, default=None)
+
+    # the last time this session was used
+    last_seen = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
+
+    # count of api calls made with this token/session (if we're updating last_seen, might as well update this too)
+    api_calls = Column(Integer, nullable=False, default=0)
+
+    # details of the browser, if available
+    # these are from the request creating the session, not used for anything else
+    ip_address = Column(String, nullable=True)
+    user_agent = Column(String, nullable=True)
```

Comment:
Unused, remove. If we ever wish to show a user from where she is logged in, we better make each client send a user-agent string as part of the grpc protocol at login time.

Reply:
I think this (along with APIs to invalidate sessions) is an important security feature to have. There have been some concerns about leftover sessions since the sessions cannot as such be destroyed once a device goes offline. It's in general a useful feature since people might travel a lot and access the site on different devices.

Anyway, the information here is stored from the user-agent and ip address of the incoming reqs, so it is filled (and not completely unused). Might also be useful at some point in debugging.

Are you sure we should remove this and implement later when we create those security features?

Reply:
I might have missed the security feature; will take a closer look tomorrow!

Reply:
You have to explain the future secure feature; do you intend to fingerprint the user and check that the fingerprint matches as well as the cookie?

Reply:
I'd just like to show the user a list of current valid sessions and allow them to invalidate them.

Reply:
Ok, that makes sense. I'm fine keeping it unused in this PR. Resolving!